ui/pages/Project/components/ProjectOverview.jsx

```diff
@@ -181,17 +181,48 @@ const mapMatchmakerStateToProps = (state, ownProps) => ({
 
 const MatchmakerOverview = connect(mapMatchmakerStateToProps)(Matchmaker)
 
+class DatasetSection extends React.PureComponent {
+
+  static propTypes = {
+    loadedSampleCounts: PropTypes.object.isRequired,
+  }
+
+  state = { showAll: false }
+
+  show = () => {
+    this.setState({ showAll: true })
+  }
+
+  render() {
+    const { loadedSampleCounts } = this.props
+    const { showAll } = this.state
+    const allLoads = Object.keys(loadedSampleCounts).sort().map(loadedDate => (
+      <div key={loadedDate}>
+        {`${new Date(loadedDate).toLocaleDateString()} - ${loadedSampleCounts[loadedDate]} samples`}
+      </div>
+    ))
+
+    const total = allLoads.length
+    if (total < 6 || showAll) {
+      return allLoads
+    }
+
+    return [
+      ...allLoads.slice(0, 2),
+      <ButtonLink key="show" padding="5px 0" onClick={this.show}>{`Show ${total - 5} additional datasets`}</ButtonLink>,
+      ...allLoads.slice(total - 3),
```

Comment:
A nitpick issue is to show the "show more" button only when it needs more lines to show all. So, if the length is 6, then it is better to show them all. So the condition for showing all should be `total <= 6`.

Reply:
no I intentionally did strict less because if the length is 6 we do not want to show them all, we only want to show the first 5. thats why we use `total - 5` when calculating the number that are hidden